File: 백준/Silver/18258. 큐 2/큐 2.java

import java.io.*;
import java.util.*;


public class Main {
    int n;

    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;
    StringBuilder sb = new StringBuilder();

    Deque<Integer> q = new ArrayDeque<>();

    void solution() throws Exception{
        n = Integer.parseInt(br.readLine());
        for(int i=0;i<n;i++){
            st=new StringTokenizer(br.readLine());
            String cmd=st.nextToken();
            switch(cmd){
                case "push":
                    int num=Integer.parseInt(st.nextToken());
                    q.offer(num);
                    break;
                case "pop":
                    if(q.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(q.poll()).append("\n");
                    break;
                case "size":
                    sb.append(q.size()).append("\n");
                    break;
                case "empty":
                    if(q.isEmpty())
                        sb.append("1\n");
                    else
                        sb.append("0\n");
                    break;
                case "front":
                    if(q.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(q.peek()).append("\n");
                    break;
                case "back":
                    if(q.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(q.getLast()).append("\n");
                    break;
            }
        }
        System.out.println(sb);
    }

    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include<iostream>
#include<queue>
#include<cstring>
#pragma warning(disable:4996)

using namespace std;

queue<int> q;

int main()
{
	int n;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		char cmd[10];
		scanf("%s",cmd);
		if (strcmp(cmd, "push") == 0) {
			int num;
			scanf("%d", &num);
			q.push(num);
		}
		else if (strcmp(cmd,"pop")==0) {
			if (q.empty()) {
				printf("-1\n");
			}
			else {
				printf("%d\n",q.front());
				q.pop();
			}
		}
		else if (strcmp(cmd, "size") == 0) {
			printf("%d\n", q.size());
		}
		else if (strcmp(cmd, "empty") == 0) {
			if (q.empty()) {
				printf("1\n");
			}
			else {
				printf("0\n");
			}
		}
		else if (strcmp(cmd, "front") == 0) {
			if (q.empty()) {
				printf("-1\n");
			}
			else {
				printf("%d\n", q.front());
			}
		}
		else {
			if (q.empty()) {
				printf("-1\n");
			}
			else {
				printf("%d\n", q.back());
			}
		}
	}

	return 0;
}